File: IRCClient/src/test/java/net/darkmeow/irc/IRCClientTest.java
package net.darkmeow.irc;

import net.darkmeow.irc.client.IRCClient;
import net.darkmeow.irc.client.data.IRCUserInfo;
import net.darkmeow.irc.client.enums.EnumPremium;
import net.darkmeow.irc.client.enums.EnumResultLogin;
import net.darkmeow.irc.client.listener.IRCClientListenableProvide;
import net.darkmeow.irc.data.ClientBrandData;
import org.junit.jupiter.api.Test;

public class IRCClientTest {

    @Test
    public void test() {
        IRCClient client = new IRCClient(new IRCClientListenableProvide() {
            @Override
            public void onUpdateUserInfo(String name, String rank, EnumPremium premium) {

            }

            @Override
            public void onMessagePublic(IRCUserInfo sender, String message) {

            }

            @Override
            public void onMessagePrivate(IRCUserInfo sender, String message) {

            }

            @Override
            public void onMessageSystem(String message) {

            }

            @Override
            public void onDisconnect(String message) {

            }
        });
        client.connect("45.207.199.139", 48088, "publicIRCTest123");
        client.login("NekoCurit", "114514", new ClientBrandData("DarkMeow", "114514", 0), result -> {
            if (result == EnumResultLogin.SUCCESS) {
                client.sendMessage("Test");
            }
        });
        try { Thread.sleep(5000); } catch (InterruptedException ignored) {}

    }
}

File: IRCClient/src/test/java/net/darkmeow/irc/IRCClientTest.kt
package net.darkmeow.irc

import net.darkmeow.irc.client.IRCClient
import net.darkmeow.irc.client.data.IRCOtherUserInfo
import net.darkmeow.irc.client.enums.EnumPremium
import net.darkmeow.irc.client.listener.IRCClientListenableProvide
import net.darkmeow.irc.client.network.IRCClientConnection
import net.darkmeow.irc.data.ClientBrandData
import net.darkmeow.irc.data.GameInfoData
import net.darkmeow.irc.network.packet.c2s.C2SPacketLogin

fun main(args: Array<String>) {
    val client = IRCClient()


    class IRCLegacy: IRCClientListenableProvide {
        override fun onMessageSystem(message: String?) {

        }

        override fun onUpdateUserInfo(name: String?, rank: String?, premium: EnumPremium?) {

        }

        override fun onMessagePublic(sender: IRCOtherUserInfo?, message: String?) {
            println(message)
        }

        override fun onMessagePrivate(sender: IRCOtherUserInfo?, message: String?) {

        }
    }

    client.listenable = IRCLegacy()
    client.connect("45.207.199.139", 48088, "publicIRCTest123")
   // readLine()
    client.login("NekoCurit", "114514", "14451", ClientBrandData("DarkMeow", "114514", 0)) {
        true
    }
    while(true) {
        readLine()?.also {
            client.sendMessage(it)
        }
    }

}